code/game/gamemodes/vampire/vampire_powers/vampire_powers.dm

```diff
@@ -0,0 +1,464 @@
+//This should hold all the vampire related powers
+/mob/living/proc/affects_vampire(mob/user)
+	//Other vampires aren't affected
+	if(mind && mind.vampire)
+		return FALSE
+	//Vampires who have reached their full potential can affect nearly everything
+	if(user && user.mind.vampire.get_ability(/datum/vampire_passive/full))
+		return TRUE
+	//Holy characters are resistant to vampire powers
+	if(mind && mind.isholy)
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/self/choose_targets(mob/user = usr)
+	perform(list(user))
+
+/obj/effect/proc_holder/spell/mob_aoe/choose_targets(mob/user = usr)
+	var/list/targets[0]
+	for(var/mob/living/L in view(range, user))
+		if(L == user)
+			continue
+		targets += L
+
+	if(!length(targets))
+		revert_cast(user)
+		return
+
+	perform(targets, user = user)
+
+/obj/effect/proc_holder/spell/proc/before_cast_vampire(list/targets)
```

Comment:
this is called by, `before_cast` which is called by `perform`. `perform` declares `user = usr` as a param. so if I was to use a parameter here, it would still come from `usr`

(if that makes any sense)

Reply:
I noticed aye. `before_cast` could also use a `user` parameter. However, this is probably outside of the PRs scope